skins/oasis/modules/OasisController.class.php

```diff
@@ -93,22 +94,23 @@ public static function onMakeGlobalVariablesScript(Array &$vars) {
 	public static function JsAtBottom(){
 		global $wgTitle;
 
-		// decide where JS should be placed (only add JS at the top for legacy datatable-based special pages)
-		return !(
-			$wgTitle->isSpecial( 'MultiLookup' ) ||
-			$wgTitle->isSpecial( 'LookupUser' ) ||
-			$wgTitle->isSpecial( 'WikiFactory' ) ||
-			$wgTitle->isSpecial( 'Listusers' ) ||
-			$wgTitle->isSpecial( 'LookupContribs' ) ||
-			$wgTitle->isSpecial( 'WhereIsExtension' ) ||
-			$wgTitle->isSpecial( 'InterwikiEdit' ) ||
-			$wgTitle->isSpecial( 'CreateBlogListingPage' ) ||
-			$wgTitle->isSpecial( 'AchievementsCustomize' )
-		);
+		// decide where JS should be placed (only add JS at the top for non-search Special and edit pages)
+		if (WikiaPageType::isSearch() || WikiaPageType::isForum()) {
+			// Remove this whole condition when AdDriver2.js is fully implemented and deployed
+
+			$jsAtBottom = true;
+		}
+		elseif ($wgTitle->getNamespace() == NS_SPECIAL || BodyController::isEditPage()) {
+			$jsAtBottom = false;
+		}
+		else {
+			$jsAtBottom = true;
+		}
+		return $jsAtBottom;
 	}
 
 	public function executeIndex($params) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This function "executeIndex" has 159 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS138)